### ISSUE TITLE
Plugin formatting still not working after installing Preview version

### ISSUE THREAD

**Issue by mbacchi** [hashicorp/vscode-terraform#1030]
## Versions

This bug is reproducible in:
 - [x] the **latest** version of the extension (below)
 - [x] the **latest** version of the language server (below)

### Extension

```
HashiCorp Terraform (Preview) v2022.3.1510
```

### Language Server

```
Starting terraform-ls 0.26.0-beta1
```

### VS Code

```
Version: 1.66.0
Commit: e18005f0f1b33c29e81d732535d8c0e47cafb0b5
Date: 2022-03-30T05:50:41.156Z
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Linux x64 5.16.18-200.fc35.x86_64
```

### Operating System

```
uname -a
Linux HOSTNAME 5.16.18-200.fc35.x86_64 #1 SMP PREEMPT Mon Mar 28 14:10:07 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
No WSL
```

### Terraform Version

```
Terraform v1.1.7
on linux_amd64
```

### Steps To Reproduce

Terraform files are not formatted when saved. I chatted with James Pogran on Twitter (https://twitter.com/ender2025/status/1505941271789113349?s=20&t=Lhe0k1qRB3rfNZ0t9i1GqA) and he suggested using the Preview version. I did this and made the other changes to the settings configuration that he recommended from here; https://github.com/hashicorp/vscode-terraform#formatting.

At first this worked, but now after maybe a week or 10 days of using the preview, I no longer have a guarantee of formatting the file when I save it. In fact at the moment it feels like it rarely if ever formats on save.

I have a current file with some resources formatted this way: 

```
  default_cache_behavior {
    allowed_methods  = var.allowed_methods
    cached_methods   = var.cached_methods
    compress         = true
    target_origin_id = local.website_origin_id
    origin_request_policy_id = aws_cloudfront_origin_request_policy.noname_service.id
```
(Notice the equal signs not lined up.)

I saved the file multiple times and reloaded vscode, and still it doesn't format correctly.

I have to run `terraform fmt` in order to format it correctly now.

After running `terraform fmt` it looks like:

```
  default_cache_behavior {
    allowed_methods          = var.allowed_methods
    cached_methods           = var.cached_methods
    compress                 = true
    target_origin_id         = local.website_origin_id
    origin_request_policy_id = aws_cloudfront_origin_request_policy.noname_service.id
```

### Expected Behavior

Formatting should be applied when the file is saved.

### Actual Behavior

Formatting of many/most files is not applied when saved.

**Comment by jpogran** [comment]
Thanks for the report, sorry you're experiencing this. We have two bug fixes landing soon that may be related to your situation:

- https://github.com/hashicorp/terraform-ls/pull/851
- https://github.com/hashicorp/terraform-ls/pull/839

But I won't be able to tell until we dig further in what's going on here. Can you print out a simplified and/or redacted folder setup you're using? For example opening `/foo/bar`, which has X number of sub folders, with `/foo/bar/../../../baz.tf` being the file you're targeting?

**Comment by mbacchi** [comment]
Thanks James. Yes the folder structure that I'm experiencing the problem today is (simplified/redacted version):

`infrastructure/terraform/foo/bar/baz/main.tf`

Let me know if I can provide more info.

**Comment by radeksimko** [comment]
@mbacchi Do you mind sharing the whole log, or at least all lines which contain `Incoming request for "textDocument/formatting"` and `Response to "textDocument/formatting"` and anything in between them?

You can find it under the `Output` pane > `HashiCorp Terraform`

<img width="253" alt="Screenshot 2022-04-11 at 10 36 37" src="https://user-images.githubusercontent.com/287584/162710929-b29d0fd5-ce11-4d21-b34a-76b761e0fee8.png">

<img width="535" alt="Screenshot 2022-04-11 at 10 35 55" src="https://user-images.githubusercontent.com/287584/162710806-bd0f20ce-8df3-44c8-a0d0-97c33b089fc8.png">

Thanks!

**Comment by radeksimko** [comment]
If you're worried about sensitive data, you may encrypt the log using [my pubkey](https://keybase.io/radeksimko) before posting it here and/or send it to radek `<at>` hashicorp.com

**Comment by mbacchi** [comment]
Thank you @radeksimko, I just saw this request and sent you a truncated log to your email. Let me know if you have further questions.

**Comment by mbacchi** [comment]
I worked with Radek on this via email to troubleshoot further. It boiled down to the fact that I had configured the plugin in settings.json according to [this](https://github.com/hashicorp/vscode-terraform#formatting) documentation, but when I started using the preview plugin that did not work because the config setting `"editor.defaultFormatter": "hashicorp.terraform"` did not match the preview plugin, as the normal `hashicorp.terraform` plugin was not installed.

Once I corrected this to use `"editor.defaultFormatter": "hashicorp.terraform-preview"` it worked, and then I was able to uninstall the preview plugin, use the standard plugin, and return the settings to use the normal plugin `"editor.defaultFormatter": "hashicorp.terraform"`.

Thanks @radeksimko for your help!
